src/qutip_cupy/dense.py

```diff
@@ -38,8 +39,8 @@ def __init__(self, data, shape=None, copy=True, dtype=cp.complex128):
         if shape is None:
             shape = base.shape
             # Promote to a ket by default if passed 1D data.
-            if len(shape) == 1:
-                shape = (shape[0], 1)
+        if len(shape) == 1:
```

Comment:
The equivalent logic in qutip's dense implementation seems a lot simpler and clearer to me. Is there some reason we can't just copy that? See https://github.com/qutip/qutip/blob/dev.major/qutip/core/data/dense.pyx#L52-L73

Reply:
mmm, I think I diverged because I did not want reshape to be called by default. But since we now hjave the `raw` constructor I think we can circumvent that.